lib/sidekiq_adhoc_job.rb

```diff
@@ -0,0 +1,38 @@
+module SidekiqAdhocJob
+
+  InvalidConfigurationError ||= Class.new(RuntimeError)
+
+  def self.configure
+    @_config = Configuration.new
+    yield @_config
+  end
+
+  def self.config
+    @_config
+  end
+
+  def self.init
+    raise InvalidConfigurationError, 'Must configure before init' unless @_config
+
+    require_relative './sidekiq_adhoc_job/services/worker_files_loader'
+
+    SidekiqAdhocJob::WorkerFilesLoader.load
+
+    Sidekiq::Web.register(SidekiqAdhocJob::Web)
+    Sidekiq::Web.tabs['adhoc_jobs'] = 'adhoc-jobs'
+    Sidekiq::Web.locales << File.expand_path('sidekiq_adhoc_job/web/locales', __dir__)
+  end
+
+  class Configuration
+    attr_accessor :worker_path_pattern, :ignore_spec
+
+    def initialize
+      @worker_path_pattern = '**/workers/**/*.rb'
+      @ignore_spec = true
+    end
+  end
+
+end
+
+Dir[File.join(File.expand_path('sidekiq_adhoc_job/utils', __dir__), '**/*.rb')].each { |file_name| require_relative file_name }
```

Comment:
Yeah traditionally the require statements go on top so you can on first glance know what are the dependencies of this file, but in this case the require statements are cluttering the file, so I put them at the bottom.

Actually I want to rewrite this part.

1. `require_relative` is wrong since I already make absolute paths here.
2. I want to rewrite the dependency graph between this main module and the worker loader module (you will realise they are depending on each other).

Reply:
https://guides.rubygems.org/patterns/#loading-code

I read a bit more about code loading and fixed a few things here cfe11d5